### PR TITLE
add documentation on --detect-loaded-modules, --allow-loaded-modules and --check-ebroot-env-vars

### PR DESCRIPTION
**Pull request by boegel**
cfr. https://github.com/hpcugent/easybuild-framework/pull/2228

preview at http://boegel-eb.readthedocs.io/en/detect_loaded_modules/Detecting_loaded_modules.html

### CODE REVIEW

**Comment by boegel** [comment]
@damianam and/or @vanzod: up for reviewing this?

File: docs/Detecting_loaded_modules.rst [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+
+Motivation
+----------
+
+Running EasyBuild in an environment where one or more (EasyBuild-generated) modules are loaded may interfere
+with the software installations performed by EasyBuild, i.e.:
+
+* they may cause installations failures, for example due to incompatibilities with the modules being loaded
+  during the installation procedure being performed;
+* they make cause installations to work in that particular environment, for example by providing a neccessary
```
Comment (vanzod) [inline docs/Detecting_loaded_modules.rst]: Do you mean: "they may" ?

File: docs/Detecting_loaded_modules.rst [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+  dependency
+
+Since manually loading modules may affect the reproducability of software installations, it should be discouraged.
```
Comment (vanzod) [inline docs/Detecting_loaded_modules.rst]: reproducability -> reproducibility

File: docs/Detecting_loaded_modules.rst [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+
+.. _detecting_loaded_modules_action_purge:
+
+``purge``: run '``module purge``' to clean environment of loaded modules
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Using ``--detect-loaded-modules=purge``, EasyBuild will run ``module purge`` if any (non-allowed) loaded modules
+are detected, in an attempt to restore the environment to a clean state before starting software installations.
+
+A short warning message are printed in case ``module purge`` was used to clean up the environment::
```
Comment (vanzod) [inline docs/Detecting_loaded_modules.rst]: "are printed" -> "is printed"

File: docs/Detecting_loaded_modules.rst [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+* :ref:`detecting_loaded_modules_EBROOT_env_vars_unset`
+* :ref:`detecting_loaded_modules_EBROOT_env_vars_warn` (current default)
+
+
+.. _detecting_loaded_modules_EBROOT_env_vars_error:
+
+``error``: report error and stop EasyBuild on unknown ``$EBROOT*`` environment variables
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When configured with ``--check-ebroot-env-vars=error``, EasyBuild will stop with a clear error message
```
Comment (vanzod) [inline docs/Detecting_loaded_modules.rst]: Missing ". " at the end

Comment (boegel) [inline docs/Detecting_loaded_modules.rst]: well, no, the sentence continues?

**Comment by boegel** [comment]
Thanks for the review @vanzod!

**Pull request merged**